texar/torch/data/data/scalar_data.py

```diff
@@ -162,7 +165,15 @@ def default_hparams():
 
     def process(self, raw_example: List[str]) -> Union[int, float]:
         assert len(raw_example) == 1
-        example: Union[int, float] = self._data_type(raw_example[0])
+
+        example_: Union[int, str]
+        if self._data_type == np.bool_:
+            example_ = int(raw_example[0])
```

Comment:
Why is it necessary to convert boolean values into `int`?

Reply:
raw_example that is read from file is of type `str`, like "0", and `np.bool("0")` returns `True`. Hence, I convert it to `int` and `np.bool(0)` returns `False`

Reply:
But it's possible to have strings like `"true"` which would fail here. I recommend using [`distutils.util.strtobool`](https://docs.python.org/3.7/distutils/apiref.html?highlight=distutils.util#distutils.util.strtobool).

Reply:
52fca5f